src/model_metric_reporter.cpp

```diff
@@ -16,138 +16,169 @@
 #include "model_metric_reporter.hpp"
 
 #include "execution_context.hpp"
+#include "metric_config.hpp"
 #include "metric_family.hpp"
 #include "metric_registry.hpp"
 
 namespace ovms {
 
-ModelMetricReporter::ModelMetricReporter(MetricRegistry* registry, const std::string& modelName, model_version_t modelVersion) :
+ModelMetricReporter::ModelMetricReporter(const MetricConfig* metricConfig, MetricRegistry* registry, const std::string& modelName, model_version_t modelVersion) :
     registry(registry) {
     if (!registry) {
         return;
     }
+
+    if (!metricConfig && !metricConfig->metricsEnabled) {
```

Comment:
This will crash if metricConfig is null, I think you meant:
```
if (!metricConfig  || (metricConfig && metricConfig->metricsEnabled)) {}
```

Write unit test for that

Reply:
Done.